monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -559,19 +559,31 @@ def get_issues():
         logger.info("Issues generated for reporting")
         return issues_dict
 
+    @classmethod
+    def get_latest_event_timestamp(cls):
+        if not cls._agent_event_repository:
+            raise RuntimeError("Agent event repository does not exist")
```

Comment:
Is this check really necessary? `ReportService.initialize()` will complain if no event repository is passed to it.

Reply:
I don't think it will be needed. The `initialize` will no complain as we have `Optional` for all repositories.

Reply:
It won't hurt anything and it will save us some debugging time if there's ever an issue. This will all be refactored (maybe next release?) anyway, so I wouldn't worry too much about it.